pkg/web/http.go

```diff
@@ -0,0 +1,69 @@
+package web
+
+import (
+	"bytes"
+	"crypto/tls"
+	"github.com/gorilla/websocket"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"time"
+)
+
+type WrappedSocketListener struct {
+	sockets chan *WrappedSocket
+	done    chan struct{}
+	addr    net.Addr
+}
+
+func (wsl *WrappedSocketListener) Accept() (net.Conn, error) {
+	select {
```

Comment:
Dont let function be called after closed. E.g.:

    if atomic.LoadInt32(&wsl.Closed) { return nil, errors.New("already closed") }